umbraco/Umbraco-CMS
# Content & media url endpoints

### Prerequisites


### Description
- New dedicated endpoints for retrieving URLs by key (e.g. for displaying picked item URLs in the content/media picker)
- Async misusage cleanup in factories

### Testing
- Create and publish some documents (preferably some in multiple cultures, don't forget to assign domains)
- Upload some temporary files and create some media from it (see example payload below)
- The following endpoints should return the relevant url info for all found documents/media by the keys you supplied `/umbraco/amanagement/api/v1/document/urls` `/umbraco/amanagement/api/v1/media/urls`

### Example payloads
Media with image cropper backing field
```
{
    "mediaType": {
        "id": "cc07b313-0843-4aa8-bbda-871c8da728c8"
    },
    "values": [
        {
            "culture": null,
            "segment": null,
            "alias": "umbracoFile",
            "value": {"src":"5344f6ae-d305-47fc-955f-3a7778b6c336"}
        }
    ],
    "variants": [
        {
            "culture": null,
            "segment": null,
            "name": "test media"
        }
    ]
}
```
Media with a generic file backing field
{
    "mediaType": {
        "id": "4c52d8ab-54e6-40cd-999c-7a5f24903e4d"
    },
    "values": [
        {
            "culture": null,
            "segment": null,
            "alias": "umbracoFile",
            "value": "5344f6ae-d305-47fc-955f-3a7778b6c336"
        }
    ],
    "variants": [
        {
            "culture": null,
            "segment": null,
            "name": "test file"
        }
    ]
}
